static/app/components/onboarding/gettingStartedDoc/onboardingLayout.tsx

```diff
@@ -116,6 +121,36 @@ export function OnboardingLayout({
     isSelfHosted,
   ]);
 
+  useEffect(() => {
+    // we need this because even though 'selectedOptions' and 'activeProductSelection'
+    // are memoized the effect still renders 2x on mount
+    if (!isMounted) {
+      return;
+    }
+
+    onProductSelectionChange?.({
+      organization: organization,
+      projectSlug,
+      platformKey,
+      projectId,
+      projectKeyId,
+      activeProductSelection,
+      platformOptions: selectedOptions,
+      api,
+    });
+  }, [
+    isMounted,
+    onProductSelectionChange,
+    organization,
+    projectSlug,
+    projectId,
+    platformKey,
+    projectKeyId,
+    activeProductSelection,
+    selectedOptions,
+    api,
+  ]);
```

Comment:
I've got 2 issues with this one.
1. It is called `onProductSelectionChange` but it is triggered for product selection and platform options.
2. Using a hook here is kinda destined to break at some point (see your comment), a change handler on the respective inputs would be nicer.

My suggestion:
Create 2 callbacks `onProductSelectionChange` and `onPlatformOptionsChange` and add a change handlers to the product selection and platform option controls.